### DISCUSSION TITLE
ELI5: How do I use this to parse my Primes output file?

### DISCUSSION THREAD

**Discussion by snarkipus** [PlummersSoftwareLLC/PrimeView#40] [Q&A] [answered]
Ok, instead of fumbling through this any further ...

I ran "make FORMATTER=json > report4.json"  and trimmed the leading lines of console output off (see attached).

The PrimeView solution builds (VS2022 Community) and successfully loads the data (which I assume is the S3Bucket data). I'll be damned if I can figure out how to point it to a local file though? FWIW, I've placed the generated report4.json file in the "...\src\Frontend\bin\Debug\net6.0\publish\wwwroot\data" directory with the other json files.

Questions:
- Modify the appsettings.json file (Solution PrimeView>Frontend>wwwroot>appsettings.json)? If so, to what? Index?
- Put the file someplace else?

Thanks for any help - I'm sure I'm missing something silly.

[report4.json.txt](https://github.com/PlummersSoftwareLLC/PrimeView/files/7790480/report4.json.txt)

**Comment by rbergen** [comment, marked as answer]
@snarkipus There are two ways to answer this question:
- Lazily: look at [JsonFileReader/README.md](https://github.com/PlummersSoftwareLLC/PrimeView/blob/main/src/JsonFileReader/README.md). It describes the way the JSON file loader works in quite some detail.
- Pragmatically: comment out/remove the `"BaseURI"` and `"IsS3Bucket"` attributes in appsettings.json.

Also make sure you add report4.json to src/Frontend/wwwroot/data instead of wwwroot/data in the publish tree (VS will still copy it there). For some reason, with VS2022, IIS Express seems to insist on loading its data from the project source wwwroot directory.

**Reply by snarkipus** [reply]
Thanks - worked as expected.

Although I did read the JsonFileReader/README.md file, something must have gone over my head. I did see that it defaulted to reading the data files in order until HTML error in the absence of the IsS3Bucket parameter being "True" ... setting it to "False" resulted in no data at all for some reason and I wasn't clear on what to do with a JSON array of names for the "Index" parameter option.

Regardless, what a cool project - I learned a ton just getting this far. Thanks for the help!

**Reply by rbergen** [reply]
> Although I did read the JsonFileReader/README.md file, something must have gone over my head.

It could also mean that the README in question is not as useful as I intended it to be...

> Regardless, what a cool project - I learned a ton just getting this far. Thanks for the help!

Thank you, I really appreciate you saying that. I had a lot of fun writing PrimeView and contributing my bit to the overall Primes project, but it's only feedback like yours that tells me if it adds something meaningful to the experience of others. :)
